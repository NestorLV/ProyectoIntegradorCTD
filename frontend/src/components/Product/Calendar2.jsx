import * as React from 'react';
import addDays from 'date-fns/addDays';
import TextField from '@mui/material/TextField';
import StaticDateRangePicker from '@mui/lab/StaticDateRangePicker';
import AdapterDateFns from '@mui/lab/AdapterDateFns';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import { makeStyles } from '@mui/styles';
//import DateRangePicker from '@mui/lab/DateRangePicker';
import Box from '@mui/material/Box';
import { createTheme, ThemeProvider } from "@mui/material/styles";

const useStyles = makeStyles({
  root: {
    borderRadius: 5,
    color: "black",
    boxShadow: '10px 5px 5px grey',
    fontSize: 20,
    background: "white",
    height: "300px",
    width: "640px",
  }
});

function getDaysAfter(date, amount) {
  return date ? addDays(date, amount) : undefined;
}

export default function Calendar2() {
  const [value, setValue] = React.useState([null, null]);
  const classes = useStyles();
  const theme = createTheme({
    palette: {
      primary: {
        main: '#ff00d4',
      },

    },
  });
  return (
    <div className={classes.root}>
      <ThemeProvider theme={theme} >
        <LocalizationProvider dateAdapter={AdapterDateFns} >
          <StaticDateRangePicker
            className={classes.root}
            calendars={window.innerWidth > 414 ? 2 : 1}
            displayStaticWrapperAs="desktop"
            maxDate={getDaysAfter(value[0], 5)}
            value={value}
            onChange={(newValue) => {
              setValue(newValue);
            }}
            renderInput={(startProps, endProps) => (
              <React.Fragment>
                <TextField {...startProps} />
                <Box sx={{ mx: 2 }}> to </Box>
                <TextField {...endProps} />
              </React.Fragment>
            )}
          />
        </LocalizationProvider>
      </ThemeProvider>

    </div>
  );
}