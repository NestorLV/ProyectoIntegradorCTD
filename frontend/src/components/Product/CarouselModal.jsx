import "./carouselModal.css"
import "react-responsive-carousel/lib/styles/carousel.min.css";
//import { Modal } from 'react-responsive-modal'
import { Carousel } from "react-responsive-carousel"
import { Modal, ModalGateway } from "react-images"


export default function CarouselModal(props){

    const closeLightbox = () => {
        props.setCurrentImage(0);
        props.setViewerIsOpen(false);
    };

    return ( 
            <div>
            <ModalGateway>
                {props.viewerIsOpen ? (
                    <Modal onClose={closeLightbox}>
                        <Carousel 
                        autoPlay={true} 
                        showIndicators={false} 
                        interval={3000} 
                        thumbWidth={140}
                        width="44vw" 
                        stopOnHover={true} 
                        emulateTouch={true} 
                        infiniteLoop={true}
                        useKeyboardArrows={true}
                       >
                           {props.images.map((image)=> {
                                return (<div>
                                            <img src={image.url} />
                                        </div>)
                                })}
                            
                        </Carousel>
                    </Modal>
                ) : null}
            </ModalGateway>
            </div>
        
    );
}

