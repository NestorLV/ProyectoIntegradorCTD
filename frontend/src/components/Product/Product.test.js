import React from "react";
import { BrowserRouter, Router } from 'react-router-dom';
import { shallow, mount } from "enzyme";
import "@testing-library/jest-dom";
import Calendar2 from "./Calendar2";
import CarouselModal from "./CarouselModal";
import DateBlock from "./DateBlock";
import DescriptionBar from "./DescriptionBar";
import FeaturesBar from "./FeaturesBar";
import ImageBar from "./ImageBar";
import InfoBar from "./InfoBar";
import Map from "./Map";
import MapBar from "./MapBar";
import Product from "./Product";
import ScoreBar from "./ScoreBar";
import Share from "./Share";
import TitleBar from "./TitleBar";
import{ createMemoryHistory } from "history"

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';


Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <Calendar2/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<Calendar2/>)
    });

    it("Deberia mostrar <Calendar2/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe("Probando el componente <CarouselModal/>", () => {
    let wrapper;
    let props={
        images:[{url:""}],
        viewerIsOpen:true,
        setCurrentImage: jest.fn(),
        setViewerIsOpen: jest.fn()
    }
   
    beforeEach(() => {
        wrapper = shallow(<CarouselModal/>)
    });

    it("Deberia mostrar <CarouselModal/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia cerrarse el carousel al hacer click en la cruz", ()=>{
        wrapper = shallow(<CarouselModal {...props}/>)
        wrapper.find("div.closeButton").simulate("close")
        expect(wrapper.prop("viewerIsOpen")).toBeFalsy()
    })
})

describe("Probando el componente <DateBlock/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<DateBlock/>)
    });

    it("Deberia mostrar <DateBlock/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe("Probando el componente <DescriptionBar/>", () => {
    let wrapper;
    let props={
        city: {
            id: 1,
            name: "ciudad",
            country: "pais",
        },
        description: "descripcion hotel"
    }
    
    beforeEach(() => {
        wrapper = shallow(<DescriptionBar {...props}/>)
    });

    it("Deberia mostrar <DescriptionBar/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
})

describe("Probando el componente <FeaturesBar/>", ()=>{
    let wrapper;
    let props={
        features: [
            { id: 1, title: "wifi", state: true },
            { id: 2, title: "pool", state: true },
        ]
    }
    
    beforeEach(() => {
        wrapper = shallow(<FeaturesBar {...props}/>)
    });

    it("Deberia mostrar <FeaturesBar/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
})

describe("Probando el componente <ImageBar/>", () => {
    let wrapper;
    let props={
        images: [
            { id: 1, title: "fotoImagen1", url: "https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612", productId: 1 },
            { id: 2, title: "fotoImagen1", url: "https://media.istockphoto.com/photos/hotel-reception-lobby-picture-id1292355630", productId: 1 },
            { id: 3, title: "fotoImagen1", url: "https://media.istockphoto.com/photos/hotel-reception-lobby-picture-id1292355630", productId: 1 },
        ],
        setViewerIsOpen:jest.fn(),
        setShareIsOpen:jest.fn()
    }

    let props2={
        images: [
            { id: 1, title: "fotoImagen1", url: "https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612", productId: 1 },
            ],
        setViewerIsOpen:jest.fn(),
        setShareIsOpen:jest.fn()
    }
    
    beforeEach(() => {
        wrapper = shallow(<ImageBar {...props}/>)
    });

    it("Deberia mostrar <ImageBar/> correctamente con varias imagenes", () => {
        wrapper.find("div.slider").simulate("click") //Simula el click para probar el if de la linea 11 si hay mas de una imagen
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia mostrar <ImageBar/> correctamente con una imagen", () => {
        wrapper = shallow(<ImageBar {...props2}/>)
        wrapper.find("div.slider").simulate("click") //Simula el click para probar el if de la linea 11 si hay solo una imagen
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia llamar a setViewerIsOpen al hacer click en el boton ver mas", ()=>{
        wrapper.find("div.verMas").simulate("click")
        expect(props.setViewerIsOpen).toHaveBeenCalled()

    })
    it("Deberia llamar a setShareIsOpen al hacer click en el icono de compartir", ()=>{
        wrapper.find("img.iconImage").first().simulate("click")
        expect(props.setShareIsOpen).toBeCalled()

    })
        
})

describe("Probando el componente <InfoBar/>", () => {
    let wrapper;
    let props = {
        rules: "normas1,normas2,normas3",
        health: "salud1,salud2,salud3",
        politics: "politicas1,politicas2,politicas3",
    }
    
    beforeEach(() => {
        wrapper = shallow(<InfoBar {...props}/>)
    });

    it("Deberia mostrar <InfoBar/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
})

describe("Probando el componente <Map/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<Map/>)
    });

    it("Deberia mostrar <Map> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe("Probando el componente <MapBar/>", () => {
    let wrapper;
    let props ={
        city: {
            id: 1,
            name: "ciudad",
            country: "pais",
        },
        latitude: 4.111784,
        longitude: -73.6396047,
    }
    
    beforeEach(() => {
        wrapper = shallow(<MapBar {...props}/>)
    });

    it("Deberia mostrar <MapBar> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})



/*describe("Probando el componente <Product/>", () => {
    let wrapper; 
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation((init) => [init, setState]);
    beforeEach(() => {
        wrapper = shallow(<Product />); 
        /*jest.mock('react-router-dom', () => ({
            useParams: jest.fn().mockReturnValue({ id: 1 }),
        }));
        
    });

    afterEach(() => {
        jest.clearAllMocks();
    });
    

    it("Deberia mostrar <Product/> correctamente", () => {
        /*const history = createMemoryHistory();
        const route = '/product/:id';
        history.push(route);

        let props = {
            goBack: history.goBack
        }
        expect(wrapper).toMatchSnapshot();
    });
})*/

describe("Probando el componente <ScoreBar/>", () => {
    let wrapper;
    let props={
        reference:"referencia",
        city: {
            id: 1,
            name: "ciudad",
            country: "pais",
        },
        qualification: 9.1,
    }

    let props2= {
        reference:"referencia",
        city: {
            id: 1,
            name: "ciudad",
            country: "pais",
        },
        qualification: 6.1,
    }

    let props3= {
        reference:"referencia",
        city: {
            id: 1,
            name: "ciudad",
            country: "pais",
        },
        qualification: 5.1,
    }


    let props4= {
        reference:"referencia",
        city: {
            id: 1,
            name: "ciudad",
            country: "pais",
        },
        qualification: 3.1,
    }

    let props5= {
        reference:"referencia",
        city: {
            id: 1,
            name: "ciudad",
            country: "pais",
        },
        qualification: 1.1,
    }

    it("Deberia mostrar <ScoreBar/> correctamente con calificacion de 9.1", () => {
        wrapper = shallow(<ScoreBar {...props}/>)
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia mostrar <ScoreBar/> correctamente con calificacion de 6.1", () => {
        wrapper = shallow(<ScoreBar {...props2}/>)
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia mostrar <ScoreBar/> correctamente con calificacion de 5.1", () => {
        wrapper = shallow(<ScoreBar {...props3}/>)
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia mostrar <ScoreBar/> correctamente con calificacion de 3.1", () => {
        wrapper = shallow(<ScoreBar {...props4}/>)
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia mostrar <ScoreBar/> correctamente con calificacion de 1.1", () => {
        wrapper = shallow(<ScoreBar {...props5}/>)
        expect(wrapper).toMatchSnapshot();
    });


})


describe("Probando el componente <Share/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<Share/>)
    });

    it("Deberia mostrar <Share/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe("Probando el componente <Titlebar/>", () => {
    let wrapper;
    let props ={
        category: {
            id: 1,
            title: "tituloCategoria",
            description: "descripcionCategoria",
            url: "imagenCategoria"
        },
        name: "nombreHotel",
    }
    beforeEach(() => {
        wrapper = shallow(<TitleBar {...props}/>)
    });

    it("Deberia mostrar <TitleBar/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})
