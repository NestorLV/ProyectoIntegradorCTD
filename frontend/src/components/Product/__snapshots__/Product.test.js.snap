// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <CarouselModal/> Deberia mostrar <CarouselModal/> correctamente 1`] = `
<div>
  <ModalGateway
    target={<body />}
  />
</div>
`;

exports[`Probando el componente <DescriptionBar/> Deberia mostrar <DescriptionBar/> correctamente 1`] = `
<div
  className="descripcionBar delimiter"
>
  <div
    className="descripcionBarChild delimiterChild"
  >
    <h2>
      Alójate en el corazón de 
      ciudad
    </h2>
    <p>
      descripcion hotel
    </p>
  </div>
</div>
`;

exports[`Probando el componente <FeaturesBar/> Deberia mostrar <FeaturesBar/> correctamente 1`] = `
<div
  className="featuresBar delimiter"
>
  <div
    className="featuresBarChild delimiterChild"
  >
    <h2>
      ¿Qué ofrece este lugar?
    </h2>
    <div
      className="icons"
    >
      <div
        className="featureBox"
        key="0"
      >
        <img
          alt="wifi"
          src="wifi.svg"
        />
        <p
          className="featureTitle"
        >
          wifi
        </p>
      </div>
      <div
        className="featureBox"
        key="1"
      >
        <img
          alt="pool"
          src="pool.svg"
        />
        <p
          className="featureTitle"
        >
          pool
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <ImageBar/> Deberia mostrar <ImageBar/> correctamente 1`] = `
<div
  className="imageBar delimiter"
>
  <div
    className="imageBarChild delimiterChild"
  >
    <div
      className="barraSup"
    >
      <img
        alt="iconSocial"
        className="iconImage"
        onClick={[Function]}
        src="iconSocial.svg"
      />
      <svg
        className="iconHeart"
        height="18"
        onClick={null}
        viewBox="0 0 25 25"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="heartColor"
          d="M12 4.248c-3.148-5.402-12-3.825-12 2.944 0 4.661 5.571 9.427 12 15.808 6.43-6.381 12-11.147 12-15.808 0-6.792-8.875-8.306-12-2.944z"
        />
      </svg>
    </div>
    <div
      className="barraInf"
    >
      <div
        className="barraIzq"
      >
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
      </div>
      <div
        className="barraDer"
      >
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <div
          className="verMas"
          onClick={[Function]}
        >
          Ver Más
        </div>
      </div>
      <div
        className="slider"
        onClick={[Function]}
      >
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <p>
          1/5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <ImageBar/> Deberia mostrar <ImageBar/> correctamente con varias imagenes 1`] = `
<div
  className="imageBar delimiter"
>
  <div
    className="imageBarChild delimiterChild"
  >
    <div
      className="barraSup"
    >
      <img
        alt="iconSocial"
        className="iconImage"
        onClick={[Function]}
        src="iconSocial.svg"
      />
      <svg
        className="iconHeart"
        height="18"
        onClick={null}
        viewBox="0 0 25 25"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="heartColor"
          d="M12 4.248c-3.148-5.402-12-3.825-12 2.944 0 4.661 5.571 9.427 12 15.808 6.43-6.381 12-11.147 12-15.808 0-6.792-8.875-8.306-12-2.944z"
        />
      </svg>
    </div>
    <div
      className="barraInf"
    >
      <div
        className="barraIzq"
      >
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
      </div>
      <div
        className="barraDer"
      >
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <div
          className="verMas"
          onClick={[Function]}
        >
          Ver Más
        </div>
      </div>
      <div
        className="slider"
        onClick={[Function]}
      >
        <img
          src="https://media.istockphoto.com/photos/downtown-cleveland-hotel-entrance-and-waiting-taxi-cab-picture-id472899538?s=612x612"
        />
        <p>
          2/5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <InfoBar/> Deberia mostrar <InfoBar/> correctamente 1`] = `
<div
  className="infoBar delimiter"
>
  <div
    className="infoBarChild delimiterChild"
  >
    <h2>
      ¿Qué tenés que saber?
    </h2>
    <div
      className="info"
    >
      <div>
        <h3>
          Normas de la casa
        </h3>
        <p
          key="0"
        >
          normas1
        </p>
        <p
          key="1"
        >
          normas2
        </p>
        <p
          key="2"
        >
          normas3
        </p>
      </div>
      <div>
        <h3>
          Salud y Seguridad
        </h3>
        <p
          key="0"
        >
          salud1
        </p>
        <p
          key="1"
        >
          salud2
        </p>
        <p
          key="2"
        >
          salud3
        </p>
      </div>
      <div>
        <h3>
          Política de Cancelación
        </h3>
        <p
          key="0"
        >
          politicas1
        </p>
        <p
          key="1"
        >
          politicas2
        </p>
        <p
          key="2"
        >
          politicas3
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <Map/> Deberia mostrar <Map> correctamente 1`] = `
<LoadScript
  googleMapsApiKey="AIzaSyDoTLld8oN8wZm0avj07nwEv0FXqdfiaOE"
  id="script-loader"
  loadingElement={
    <p>
      Cargando
    </p>
  }
  version="weekly"
>
  <GoogleMap
    center={
      Object {
        "lat": undefined,
        "lng": undefined,
      }
    }
    mapContainerStyle={
      Object {
        "height": "400px",
        "width": "100%",
      }
    }
    onClick={[Function]}
  >
    <Marker
      onClick={[Function]}
      position={
        Object {
          "lat": undefined,
          "lng": undefined,
        }
      }
    />
  </GoogleMap>
</LoadScript>
`;

exports[`Probando el componente <MapBar/> Deberia mostrar <MapBar> correctamente 1`] = `
<div
  className="mapBar delimiter"
>
  <div
    className="mapBarChild delimiterChild"
  >
    <h2>
      ¿Dónde vas a estar?
    </h2>
    <p>
      ciudad , pais
    </p>
    <div
      className="map"
    >
      <Memo(Map)
        latitude={4.111784}
        longitude={-73.6396047}
        zoom={15}
      />
    </div>
  </div>
</div>
`;

exports[`Probando el componente <ScoreBar/> Deberia mostrar <ScoreBar/> correctamente con calificacion de 1.1 1`] = `
<div
  className="scoreBar delimiter"
>
  <div
    className="scoreBarChild delimiterChild"
  >
    <div
      className="barraIzq"
    >
      <p>
         
        <img
          alt="iconLocation"
          className="iconLocation"
          src="iconLocation.svg"
        />
          
         ciudad, pais
      </p>
      <p
        className="scoreBarReference"
      >
        referencia
      </p>
    </div>
    <div
      className="barraDer"
    >
      <div
        className="calificacionLetra"
      >
        <p>
          Malo
        </p>
        <div
          className="calificacionEstrellas"
        >
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
        </div>
      </div>
      <p
        className="calificacionEntero"
      >
        1
      </p>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <ScoreBar/> Deberia mostrar <ScoreBar/> correctamente con calificacion de 3.1 1`] = `
<div
  className="scoreBar delimiter"
>
  <div
    className="scoreBarChild delimiterChild"
  >
    <div
      className="barraIzq"
    >
      <p>
         
        <img
          alt="iconLocation"
          className="iconLocation"
          src="iconLocation.svg"
        />
          
         ciudad, pais
      </p>
      <p
        className="scoreBarReference"
      >
        referencia
      </p>
    </div>
    <div
      className="barraDer"
    >
      <div
        className="calificacionLetra"
      >
        <p>
          Regular
        </p>
        <div
          className="calificacionEstrellas"
        >
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
        </div>
      </div>
      <p
        className="calificacionEntero"
      >
        3
      </p>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <ScoreBar/> Deberia mostrar <ScoreBar/> correctamente con calificacion de 5.1 1`] = `
<div
  className="scoreBar delimiter"
>
  <div
    className="scoreBarChild delimiterChild"
  >
    <div
      className="barraIzq"
    >
      <p>
         
        <img
          alt="iconLocation"
          className="iconLocation"
          src="iconLocation.svg"
        />
          
         ciudad, pais
      </p>
      <p
        className="scoreBarReference"
      >
        referencia
      </p>
    </div>
    <div
      className="barraDer"
    >
      <div
        className="calificacionLetra"
      >
        <p>
          Bueno
        </p>
        <div
          className="calificacionEstrellas"
        >
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
        </div>
      </div>
      <p
        className="calificacionEntero"
      >
        5
      </p>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <ScoreBar/> Deberia mostrar <ScoreBar/> correctamente con calificacion de 6.1 1`] = `
<div
  className="scoreBar delimiter"
>
  <div
    className="scoreBarChild delimiterChild"
  >
    <div
      className="barraIzq"
    >
      <p>
         
        <img
          alt="iconLocation"
          className="iconLocation"
          src="iconLocation.svg"
        />
          
         ciudad, pais
      </p>
      <p
        className="scoreBarReference"
      >
        referencia
      </p>
    </div>
    <div
      className="barraDer"
    >
      <div
        className="calificacionLetra"
      >
        <p>
          Muy Bueno
        </p>
        <div
          className="calificacionEstrellas"
        >
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
        </div>
      </div>
      <p
        className="calificacionEntero"
      >
        6
      </p>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <ScoreBar/> Deberia mostrar <ScoreBar/> correctamente con calificacion de 9.1 1`] = `
<div
  className="scoreBar delimiter"
>
  <div
    className="scoreBarChild delimiterChild"
  >
    <div
      className="barraIzq"
    >
      <p>
         
        <img
          alt="iconLocation"
          className="iconLocation"
          src="iconLocation.svg"
        />
          
         ciudad, pais
      </p>
      <p
        className="scoreBarReference"
      >
        referencia
      </p>
    </div>
    <div
      className="barraDer"
    >
      <div
        className="calificacionLetra"
      >
        <p>
          Excelente
        </p>
        <div
          className="calificacionEstrellas"
        >
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 15 13"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
              fill="#1DBEB4"
            />
          </svg>
        </div>
      </div>
      <p
        className="calificacionEntero"
      >
        9
      </p>
    </div>
  </div>
</div>
`;

exports[`Probando el componente <Share/> Deberia mostrar <Share/> correctamente 1`] = `
<ForwardRef
  center={true}
  onClose={[Function]}
>
  <div
    className="containerSharePrincipal"
  >
    <h3>
      Compartir
    </h3>
    <div
      className="containerShare"
    >
      <div
        className="shareBlock"
      >
        <div
          className="shareItem"
        >
          <a
            href="mailto:?Subject=Mira este alojamiento localhost:3000/product/undefined"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="logo email"
              src="email.png"
            />
          </a>
          <p>
            Correo
          </p>
        </div>
        <div
          className="shareItem"
        >
          <a
            href="https://www.facebook.com/sharer/sharer.php?u=[localhost:3000/product/undefined]"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="logo facebook"
              src="facebook.svg"
            />
          </a>
          <p>
            Facebook
          </p>
        </div>
      </div>
      <div
        className="shareBlock"
      >
        <div
          className="shareItem"
        >
          <a
            href="https://twitter.com/intent/tweet?text=[Mira este producto ]&url=[localhost:3000/product/undefined]&hashtags=[GranAlojamiento]"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="logo twitter"
              src="twitter.svg"
            />
          </a>
          <p>
            Twitter
          </p>
        </div>
        <div
          className="shareItem"
        >
          <a
            href="https://api.whatsapp.com/send?text=[Mira este alojamiento: localhost:3000/product/undefined]"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="logo whatsapp"
              src="whatsapp.svg"
            />
          </a>
          <p>
            Whatsapp
          </p>
        </div>
      </div>
    </div>
  </div>
</ForwardRef>
`;

exports[`Probando el componente <Titlebar/> Deberia mostrar <TitleBar/> correctamente 1`] = `
<div
  className="titleBar delimiter"
>
  <div
    className="titleBarChild delimiterChild"
  >
    <div>
      <p>
        <Component />
      </p>
      <h2>
        nombreHotel
      </h2>
    </div>
    <div>
      <img
        alt="IconLocation"
        className="iconLocation"
        src="iconBack.svg"
      />
    </div>
  </div>
</div>
`;
